docs/csharp/debugging.md

```diff
@@ -166,6 +169,45 @@ The following tokens are also supported in the log message:
+
+### Supported .NET apps and scenarios
+
+C# Dev Kit supports the "classic" Hot Reload experience, also known as Edit and Continue. You can apply code changes while debugging regardless if you are stopped at a breakpoint or the program is running.
+
+As of November 2023, some features such as `MetadataUpdateHandler`, which enables ASP.NET Core applications to automatically refresh the browser [after a change is made](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.metadata.metadataupdatehandlerattribute?view=net-8.0), are not available yet. Applying code changes without debugging is also not supported.
+
+The runtime added support for applying changes while debugging on Linux/Mac on .NET 8, so a runtime version of .NET 8+ is required when applying code changes for .NET apps running on these operating systems.
+
+| Application type | Supports Hot Reload with C# Dev Kit | .NET 8+ Required |
+|------------------|-----------|-----------|
+| Console | ✅ | Linux/Mac Only |
+| Test Projects | ✅ | Linux/Mac Only |
+| Class Library Projects | ✅ | Linux/Mac Only |
+| ASP.NET Core | ❌* _Currently only supports changes on `.cs` files_ | Linux/Mac Only |
+| MAUI | ❌* _Available soon_ | -- |
+| Unity | ❌ | -- |
+
+See [supported projects](/docs/csharp/cs-dev-kit-faq#what-project-types-are-currently-supported) for more information on projects currently supported by C# Dev Kit. Also see the [C# Dev Kit FAQ](/docs/csharp/cs-dev-kit-faq#hot-reload) for more information on troubleshooting other unsupported scenarios.
+
+### How to apply code changes
+
+Once a Hot Reload session starts and new changes are made, you can apply the changes to the application with any of the following actions:
+
+| Action                                                 | Explanation                                                                                                                                                            |
+|--------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Hot Reload <br> `Ctrl+Shift+Enter` | Available from the **Debug Toolbar** and will start applying code changes. |
+| Save File <br> `kb(workbench.action.files.save)`        | Start applying code changes if `csharp.debug.hotReloadOnSave` is set to true. See [user settings](#user-settings) for more information. |
+| Continue / Step Over / Step Into / Step Out <br> `kb(workbench.action.debug.continue)` / `kb(workbench.action.debug.stepOver)` / `kb(workbench.action.debug.stepInto)` / `kb(workbench.action.debug.stepOut)`        | When changes were made while on a break state (for example, while stopped at a breakpoint), these commands will automatically them. |
+
+![Hot Reload demonstrated on ASP.NET](images/debugging/hotreload-demo.gif)
```

Comment:
```suggestion
![Hot Reload demonstrated on ASP.NET](images/debugging/hotreload-demo.gif)
```

Hmm, it's a bit strange that we say about that hot reload isn't really supported yet in ASP.NET Core apps, but then we show a demo of it partially working. Is there a different demo that we can show?

Reply:
I couldn't think of something better that shows UI changes targeting only `.cs` files... It seemed better than a console app.

Reply:
We'll leave as it is for now